docs/GenericsManifesto.md

```diff
@@ -249,6 +264,21 @@ typealias AnyObject = protocol<class>
 
 See the "Existentials" section, particularly "Generalized existentials", for more information.
 
+### Generalized supertype constraints
+
+Currently, supertype constraints may only be specified using a concrete class or protocol type.  This prevents us from abstracting over the supertype.
+
+```Swift
+protocol P {
+  associatedtype Base
+  associatedtype Derived: Base
+}
+```
+
+In the above example `Base` may be any non-final class or existential type.  If `Base` is a class `Derived` must inherit from `Base`.  If `Base` is an existential `Derived` may be any type that provides a compatible conformance, including a value type.
```

Comment:
I had assumed that we were talking about a superclass constraint here, but I *think* it generalizes to existentials correctly. It's slightly bothersome to have the requirement on `Base` be "non-final class or existential type", and I wonder if perhaps there's a looser definition that's okay. For example, maybe `Base == Derived` is always okay, even if `Base` is (say) a value type.

Reply:
That's good feedback.  I didn't word this precisely enough.  I think it makes sense to allow `Base` to be any type.  `Derived` may be the same as `Base`, a subclass of `Base` when `Base` is a non-final class or a conforming type when `Base` is an existential.  Does that loosening make sense?

Reply:
Yes, that makes sense to me.

Reply:
Would it be clearer to refer to these as "is" constraints? Derived may be any type so long as values of that type return true from a `is Base` condition.